app/src/main/java/com/example/hungrywolfs/adapters/MealAdapter.kt

```diff
@@ -36,6 +40,10 @@ class MealAdapter :
                 error(R.drawable.ic_broken_image)
             }
         }
+        holder.mealConstraintLayout.setOnClickListener {
+            idMealResult = meals[position].idMeal
```

Comment:
You don't need the idMealResult, just call the clickListener with the right-hand side.